contracts/strategy/LiquidationStrategy.sol

```diff
@@ -438,9 +505,114 @@ contract LiquidationStrategy is BaseStrategy, ReentrancyGuard, IFlashLendingCall
             uint256 _tokencoinBalanceAfter = ERC20(_tokencoinAddress).balanceOf(address(this));
 
             receivedAmount = _tokencoinBalanceAfter.sub(_tokencoinBalanceBefore);
+    }
+
+    function _sellCollateralV3(
+        address _tokenIn,
+        address _tokenOut,
+        uint256 _amountIn,
+        uint24 _poolFee,
+        address _routerV3
+    ) internal returns (uint256 receivedAmount) {
+        uint256 _tokencoinBalanceBefore = ERC20(_tokenOut).balanceOf(address(this));
+
+        (bytes memory commands, bytes[] memory inputs) = _encodeSwapExactInputSingle(
+            address(this),
+            _amountIn,
+            0,
+            abi.encodePacked(_tokenIn, _poolFee, _tokenOut),
+            true
+        );
+
+        ERC20(_tokenIn).safeApprove(uniswapV3Info[_routerV3].permit2, _amountIn);
```

Comment:
we read `uniswapV3Info[_routerV3].permit2` from storage 3 times in this function, we can just store the value in a local variable

Reply:
on top of that this is is called internally from the functions that already loaded permit2 value, so we can pass it as param